src/service/daemon.js

```diff
@@ -23,15 +23,10 @@ const Gtk = imports.gi.Gtk;
 
 // Bootstrap
 function get_datadir() {
-    const m = /@([^:]+):\d+/.exec(new Error().stack.split('\n')[1]);
-    let path = null;
-    const symbolic_link = Gio.File.new_for_path(m[1])
-        .query_info('standard::*', null, null)
-        .get_symlink_target();
-    if (symbolic_link)
-        path = symbolic_link;
-    else
-        path = m[1];
+    let [, path] = /@([^:]+):\d+/.exec(new Error().stack.split('\n')[1]);
+    const info = Gio.File.new_for_path(path)
+        .query_info('standard::*', Gio.FileQueryInfoFlags.NOFOLLOW_SYMLINKS, null);
+    path = info.get_is_symlink() ? info.get_symlink_target() : path;
 
     return Gio.File.new_for_path(path).get_parent().get_parent().get_path();
```

Comment:
Doubt it's a big deal either way, but you could save yourself a `Gio.File` instantiation in the (presumably-typical) non-symlink situation, this way:

```suggestion
    const [, path] = /@([^:]+):\d+/.exec(new Error().stack.split('\n')[1]);
    let file = Gio.File.new_for_path(path);
    const info = file.query_info(
        'standard::*', Gio.FileQueryInfoFlags.NOFOLLOW_SYMLINKS, null);
    if (info.get_is_symlink()) {
        file = Gio.File.new_for_path(info.get_symlink_target());
    }
    return file.get_parent().get_parent().get_path();
```